### ISSUE TITLE
Testvoter allows all entities for access

### ISSUE THREAD

**Issue by alexander-schranz** [sulu/sulu#2756]
| Q | A |
| --- | --- |
| Bug? | no |
| New Feature? | no |
#### Actual Behavior

Currently its not possible to create security tests based on voters because the test voter allow all entities. In my case e.g. only the creator of the entity is allowed to edit a entity for this I did write a test case which is not working because the TestVoter allow the access.

https://github.com/sulu/sulu/blob/develop/src/Sulu/Bundle/TestBundle/Testing/TestVoter.php

**Comment by danrot** [comment]
Why don't you deactivate the test voter then for this particular bundle? What the voter does is need in almost all other tests. Or what exactly do you want? Do you want to have a blacklist or something?

**Comment by alexander-schranz** [comment]
How can I deactivate the testvoter? In my case its a functional test on the website and the test kernel is based on the WebsiteKernel.

I currently did change the strategy for tests but its more a hack because then it will also not work if I have multiple voters:

```
security:
    access_decision_manager:
        strategy: unanimous
```

I think the testuser should just have the basic permissions for the sulu tests.

**Comment by danrot** [comment]
That adds even more complexity to the tests 😕 Without looking at it any further it would probably be possible to add a compiler pass and remove it somehow. That should be the way to go IMO, because otherwise the `TestBundle` containing the `TestVoter` must know about every other bundle.

**Comment by alexander-schranz** [comment]
fixed in #4748
